### ISSUE TITLE
aria-required-children violation for tablist role

### ISSUE THREAD

**Issue by mturnwall** [dequelabs/axe-core#3915]
### Product

axe-core

### Question

I'm trying to figure out why the following HTML structure is giving me a `aria-required-children` violation.

```
<ul role="tablist" class="chrome-tabs-group" aria-owns="tab_1">
    <li id="li_tab_1" role="presentation" class="chrome-one-tab-container" data-tooltip="record" style="width: 180px">
        <button draggable="true" role="tab" id="tab_1" class="chrome-one-tab -meta is-selected has-children move-cursor" tabindex="0" aria-grabbed="false" aria-selected="true" aria-label="record">
            <div class="chrome-one-tab-content has-presence">
                <span class="presence-dot is-hidden"></span>
            </div>
            <div class="chrome-one-tab-content">
                <span class="chrome-one-tab-label" data-truncation="true">record</span>
            </div>
        </button>
        <button type="button" class="chrome-one-tab-close" aria-label="Close Tab record" data-tooltip="Close Tab record">
            <now-icon component-id="close-tab-tab_1" icon="close-outline" now-id="close-tab-tab_1" dir="ltr"></now-icon>
        </button>
    </li>
</ul>

```

I tried moving the `tab` role to the `li` but then I get the `nested-interactive` violation which is to be expected. I also tried adding the `aria-owns` attribute to the `ul`. I'm not sure what the actual issue.

Here is a screenshot of the violation. `sn-canvas-tabs` is the parent element and is a custom web component.

![image](https://user-images.githubusercontent.com/44970/219467156-4057304b-f04f-49bb-8fca-df0b4f57f3b3.png)

**Comment by mturnwall** [comment]
I noticed a similar structure recommended here #2415

**Comment by straker** [comment]
Thanks for the issue. It looks like the reason it is failing is because the second `button` element is not `role=tab` and all children of a `tablist` must be a `tab` role. Because of that the rule is failing with the message `Element has children which are not allowed`

**Comment by mturnwall** [comment]
Thank you!

**Comment by tesk9** [comment]
@straker could you please provide a reference for where it's specified that all children of a `tablist` must be `tab` role? From what I see in the spec, at least one element owned or contained by the `tablist` must be a `tab`. I don't see guidance that no other elements are allowed.

I am aware that there's work underway to clarify required roles in https://github.com/w3c/aria/issues/1033, but as far as I understand, the current language doesn't disallow the tablist from containing children/descendants with other roles.

> Authors MUST ensure [elements](https://www.w3.org/TR/wai-aria-1.1/#dfn-element) with [role](https://www.w3.org/TR/wai-aria-1.1/#dfn-role) tab are contained in, or [owned](https://www.w3.org/TR/wai-aria-1.1/#dfn-owned-element) by, an element with the role tablist.

From [ARIA 1.1 tab section](https://www.w3.org/TR/wai-aria-1.1/#tab)

> An 'owned element' is any DOM descendant of the [element](https://www.w3.org/TR/wai-aria-1.1/#dfn-element), any element specified as a child via [aria-owns](https://www.w3.org/TR/wai-aria-1.1/#aria-owns), or any DOM descendant of the owned child.

[owned element definition](https://www.w3.org/TR/wai-aria-1.1/#dfn-owned-element)

---

Concretely, sometimes our tabs render with an icon and a tooltip rather than plain text. Our tooltips are in the DOM next to the element that they're related to, which in this instance means inside the element with role tablist.
